### ISSUE TITLE
Make an example of running a process

### ISSUE THREAD

**Issue by chrisdone** [haskell-lang/haskell-lang#26]
Experience shows us that clients and newbie Haskellers generally need to figure out how to launch a process, it's not obvious. So let's include an example out of the box for that which uses the `process` package.

**Comment by bitemyapp** [comment]
Sounds like a good idea. Go under /package/process/tutorial or some-such?

**Comment by chrisdone** [comment]
Indeed!

**Comment by bitemyapp** [comment]
@chrisdone I think you've highlighted something important as a lot of users are going to be looking for the ease of use langs like Ruby have for running a process and getting the stdout.

**Comment by chrisdone** [comment]
@bitemyapp Yeah, one of our clients actually asked for an example code snippet for this use-case (and is of a Ruby background to boot), seems both of our experiences echo this need.

**Comment by bitemyapp** [comment]
@chrisdone when I worked in ops at a webhost, the usual upgrade path for sysadmins learning to code was bash -> Ruby because Ruby was still a "real" language with libraries, but it was relatively trivial to just run a process and collect results.

One issue with `process` is that there's a bit of a soup of little distinguished options:

http://hackage.haskell.org/package/process-1.4.2.0/docs/System-Process.html

There's a hint in the Haddocks that `createProcess` is probably what you want, but things quickly get confusing after that.

**Comment by psibi** [comment]
Compared to most of the haddocks in the Haskell ecosystem, I find that `createProcess` has been well documented. Even some partial examples has been given based on that function.

**Comment by chrisdone** [comment]
It's well-documented, it's just not a very satisfying API.

**Comment by bitemyapp** [comment]
@psibi well, it gives people an out/obvious win right? I give it credit for that, that means some people at least are going to be able to get the basic case working without a lot of puzzling.

**Comment by snoyberg** [comment]
Once #39 is implemented, I'd recommend we have a full blown tutorial for processes, starting with a good basic example, and delving into more complex use cases. I'd also recommend that we cover related libraries like streaming-commons and process-extras (which both have better binary support).
